app/src/main/java/org/openobservatory/ooniprobe/test/suite/AbstractSuite.java

```diff
@@ -133,8 +133,9 @@ public ArrayList<AbstractSuite> asArray(){
 		return list;
 	}
 
-	public boolean isTestEmpty() {
-		return testList.length == 0;
+	public boolean isTestEmpty(PreferenceManager preferenceManager) {
+		getTestList(preferenceManager);
+		return testList == null || testList.length == 0;
```

Comment:
Do I understand it correctly that the bugfix is the above line and the rest of the diff is work to make sure we're always going to call `isTestEmpty` with the preference manager?

Can you explain why do we need to call `getTestList(preferenceManager)` here? Is this a function that mutates the state of the `AbstractSuite`? (I assume `testList` is then an attribute?)

Reply:
`testList` will remain empty until `getTestList(preferenceManager)` is called at least once, allowing the subclass to add the component tests based on preferences to the `testList`.

Reply:
I see, thank you!